gateway/app/interfaces/qodProvisioning.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+
+from app.schemas.afSessionWithQos import UserPlaneNotificationData
+from app.schemas.qodProvisioning import (
+    CreateProvisioning,
+    ProvisioningInfo,
+    RetrieveProvisioningByDevice,
+)
+
+
+class ResourceNotFound(Exception):
+    pass
+
+
+class DeviceNotFound(Exception):
+    pass
+
+
+class QoDProvisioningAbstractInterface(ABC):
```

Comment:
The abstract is redundant here

```suggestion
class QoDProvisioningInterface(ABC):
```